src/Middleware/Localization/test/UnitTests/AcceptLanguageHeaderRequestCultureProviderTest.cs

```diff
@@ -17,141 +18,177 @@ public class AcceptLanguageHeaderRequestCultureProviderTest
         [Fact]
         public async Task GetFallbackLanguage_ReturnsFirstNonNullCultureFromSupportedCultureList()
         {
-            var builder = new WebHostBuilder()
-                .Configure(app =>
+            using var host = new HostBuilder()
+                .ConfigureWebHost(webHostBuilder =>
                 {
-                    app.UseRequestLocalization(new RequestLocalizationOptions
+                    webHostBuilder
+                    .UseTestServer()
+                    .Configure(app =>
                     {
-                        DefaultRequestCulture = new RequestCulture("en-US"),
-                        SupportedCultures = new List<CultureInfo>
+                        app.UseRequestLocalization(new RequestLocalizationOptions
                         {
-                            new CultureInfo("ar-SA"),
-                            new CultureInfo("en-US")
-                        }
-                    });
-                    app.Run(context =>
-                    {
-                        var requestCultureFeature = context.Features.Get<IRequestCultureFeature>();
-                        var requestCulture = requestCultureFeature.RequestCulture;
-                        Assert.Equal("ar-SA", requestCulture.Culture.Name);
-                        return Task.FromResult(0);
+                            DefaultRequestCulture = new RequestCulture("en-US"),
+                            SupportedCultures = new List<CultureInfo>
+                            {
+                                new CultureInfo("ar-SA"),
+                                new CultureInfo("en-US")
+                            }
+                        });
+                        app.Run(context =>
+                        {
+                            var requestCultureFeature = context.Features.Get<IRequestCultureFeature>();
+                            var requestCulture = requestCultureFeature.RequestCulture;
+                            Assert.Equal("ar-SA", requestCulture.Culture.Name);
+                            return Task.FromResult(0);
+                        });
                     });
-                });
+                }).Build();
 
-            using (var server = new TestServer(builder))
+            await host.StartAsync();
+
+            using (var server = host.GetTestServer())
             {
                 var client = server.CreateClient();
                 client.DefaultRequestHeaders.AcceptLanguage.ParseAdd("jp,ar-SA,en-US");
                 var count = client.DefaultRequestHeaders.AcceptLanguage.Count;
                 var response = await client.GetAsync(string.Empty);
                 Assert.Equal(3, count);
             }
+
+            await host.StopAsync();
```

Comment:
Remind me why you had to call stop? Why isn't Dispose good enough?

Reply:
I don't remember exactly, I think this was the issue where memory blocks were not being disposed which breaks in debug.

Reply:
That's only an issue in Kestrel and SignalR (?), not TestServer.

Reply:
Ah true, only if you use Kestrel. Want me to remove the `StopAsync`'s then?

Reply:
Remove a few just to confirm it works without them.

Reply:
Seems to work fine. I'll remove them.